ros/src/tl_detector/tl_detector.py

```diff
@@ -95,9 +99,21 @@ def image_cb(self, msg):
         if not self.waypoints_2d or not self.waypoint_tree:
             return
 
-        t = rospy.get_rostime()
-        if int(t.nsecs * 1e-8) % 5 != 0:
-            return
+        # check if it's time to process the next image
+        timestamp_now = rospy.get_rostime()
+        if (self.timestamp_before == None):
+            # it's the very first image
+            self.timestamp_before = timestamp_now
+            self.skipping_duration = rospy.Duration(SKIPPING_DURATION)
+        else:
+            time_elapsed = timestamp_now - self.timestamp_before
+            if (time_elapsed < self.skipping_duration):
+                # do not process this image, wait until enough time has elapsed
+                return
+            else:
+                # set current timestamp and next duration, take jitter into account
+                self.timestamp_before = timestamp_now
+                self.skipping_duration += rospy.Duration(SKIPPING_DURATION) - time_elapsed
```

Comment:
Why is this a +=? Then the duration is increasing as the car runs?

Actually, wouldn't it be simpler just to delete this line and keep the skip at 0.5s indefinitely? I'm guessing that camera images are coming in at a stable enough rate that we don't have to worry about minor differences ("jitter," as you say).

Reply:
Never mind. I read this over again, and the math is correct.